.github/workflows/security-audit.yml

```diff
@@ -0,0 +1,24 @@
+name: Security audit
```

Comment:
should this be part of https://github.com/deislabs/akri/blob/main/.github/workflows/check-rust.yml?

Reply:
I thought to group all security audits in one workflow and later we can expand it to include other security checks too. No strong opinion thou, I can move it to check-rust.yml since they run on the same triggers.